Automattic/wp-calypso
# Reader post sharing / Press this - replace created post content with an embed of the original

Resolves item 3 of pe7F0s-P6-p2

## Proposed Changes

* This replaces the content of the post that is started from sharing a post from the reader/press-this with an embed of the original post. 
After:
<img width="500" alt="Screenshot 2023-05-10 at 11 43 51 AM" src="https://github.com/Automattic/wp-calypso/assets/28742426/535e5856-b1e7-4d75-91f5-fb629ce1b9aa">

## Notes on the Embed block

(EDIT- these issues have all been resolved) Working on this seems to be highlighting some issues with the core embed block for WP sites. Fixing these is out of scope of this PR, but seems like they may be a blocker for making these changes. Here are 3 examples of using the embed block in the editor and how it is rendered on the front end:

### In the editor:
<img width="500" alt="Screenshot 2023-05-10 at 12 12 25 PM" src="https://github.com/Automattic/wp-calypso/assets/28742426/d38fada6-d8c8-4ca1-af0f-de8c81420c7f">

* The first and third embed blocks are cut-off
* The second (middle) embed block only renders a link.

### On the front-end:
<img width="500" alt="Screenshot 2023-05-10 at 12 09 50 PM" src="https://github.com/Automattic/wp-calypso/assets/28742426/f04890b7-9902-48c5-8f51-e124d36ec41d">

* The first and third embed blocks are still cut-off, however the third is less-so than before. Buttons for comments and sharing are still cut off on the first.
* The second (middle) that was only a link in the editor ends up rendering as expected on the front-end.

## Testing Instructions



* Build and sync this branch of `wpcom-block-editor` to your sandbox.  ( PCYsg-l4k-p2 for info on this step for wpcom-block-editor )
* sandbox the public-api, widgets, and site you will share the post from.
* Build this calypso branch (or similarly use the calypso.live link supplied below).
* Go to the reader.
* Find a post including a featured image.
* Use the sharing (curved arrow) icon to open the sharing popover.
* select the site you would like to share this from.
* <img width="300" alt="Screenshot 2023-05-09 at 1 40 49 PM" src="https://github.com/Automattic/wp-calypso/assets/28742426/42a9afd4-6e62-41eb-9bc8-80f15af677c6">
* when the editor loads, verify the only post content is an embed of the original post.

## Pre-merge Checklist
